code/modules/reagents/chemistry/reagents/medicine.dm

```diff
@@ -1289,11 +1294,28 @@
 				return ..()
 			else
 				for(var/obj/item/organ/external/E in M.bodyparts)
-					if(E.is_broken())
-						if(prob(50)) // Each tick has a 50% chance of repearing a bone.
+					if(prob(25)) // Each tick has a 50% chance of repearing a bone.
+						if(E.status & (ORGAN_INT_BLEEDING | ORGAN_BROKEN | ORGAN_SPLINTED | ORGAN_DEAD)) //I can't just check for !E.status
 							to_chat(M, "<span class='notice'>You feel a burning sensation in your [E.name] as it straightens involuntarily!</span>")
 							E.rejuvenate() //Repair it completely.
-							break
+				if(ishuman(M))
+					var/mob/living/carbon/human/H = M
+					for(var/obj/item/organ/internal/I in M.internal_organs) // 60 healing to all internal organs.
+						I.heal_internal_damage(4)
+					if(H.blood_volume < BLOOD_VOLUME_NORMAL * 0.7)// If below 70% blood, regenerate 150 units total
+						H.blood_volume += 10
+					for(var/datum/disease/critical/heart_failure/HF in H.viruses)
```

Comment:
Maybe treat shock with this as well? Just so they don't degrade into it if they get unlucky.

Reply:
healing will fix it, as shock cures above 25 hp